crypto/asn1/x_int64.c

```diff
@@ -52,8 +52,11 @@ static int uint64_i2c(ASN1_VALUE **pval, unsigned char *cont, int *putype,
         && utmp == 0)
         return -1;
     if ((it->size & INTxx_FLAG_SIGNED) == INTxx_FLAG_SIGNED
-        && (int64_t)utmp < 0)
+        && (int64_t)utmp < 0) {
+        /* i2c_uint64_int() assumes positive values */
+        utmp = (uint64_t)(-(int64_t)utmp);
```

Comment:
This will probably produce a warning in VS, you probably want: `utmp = (uint64_t)(0-(int64_t)utmp);`